packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/stats_table.dart

```diff
@@ -142,32 +142,47 @@ class _StatsTableState extends State<StatsTable> {
       _RetainedSizeColumn(),
     ];
 
-    if (!widget.sorting.initialized) {
-      widget.sorting
+    if (!_item.sorting.initialized) {
+      _item.sorting
         ..direction = SortDirection.descending
         ..columnIndex = _columns.indexOf(_shallowSizeColumn)
         ..initialized = true;
     }
   }
 
+  HeapStatsRecord? get _selectedRecord =>
+      _item.statsToShow.recordsByClass[widget.controller.selectedClass.value];
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    _selectionNotifier.value = _selectedRecord;
+    addAutoDisposeListener(
+      widget.controller.selectedClass,
+      () => _selectionNotifier.value = _selectedRecord,
+    );
+  }
+
   @override
   Widget build(BuildContext context) {
     return FlatTable<HeapStatsRecord>(
       columns: _columns,
-      data: widget.data.records,
+      data: _item.statsToShow.records,
       keyFactory: (e) => Key(e.heapClass.fullName),
-      onItemSelected: (r) => widget.selectedRecord.value = r,
-      selectionNotifier: widget.selectedRecord,
-      sortColumn: _columns[widget.sorting.columnIndex],
-      sortDirection: widget.sorting.direction,
+      onItemSelected: (r) =>
+          widget.controller.setSelectedClass(r.heapClass.fullName),
+      selectionNotifier: _selectionNotifier,
```

Comment:
we should be accessing a value listenable from the controller here, not storing a value notifier in this widget state